docs/source/tutorial/extract.rst

```diff
@@ -0,0 +1,432 @@
+    from kf_lib_data_ingest.etl.extract.operations import *
+    from kf_lib_data_ingest.etl.transform.standard_model.concept_schema import (
+        CONCEPT
+    )
+
+It's a Python module! Cool! That lets us do all kinds of neat stuff like
+import predefined constants and functions.
+
+Fetching the data
+-----------------
+
+.. code-block:: python
+
+    source_data_url = 'file://path/to/family_and_phenotype.tsv'
+
+The first thing that the extractor does for every config file is fetch the
+related source data. This specifies where the file lives so that the code can
+fetch it.
+
+Supported protocol prefixes are:
+``file://``, ``s3://``, ``http://``, ``https://``
+
+Loading the data
+----------------
+
+.. code-block:: python
+
+    source_data_loading_parameters = {}
+
+This file contains tab-separated values with a simple layout where the first
+row is the column headers, so it gets loaded correctly by default. With more
+complex files we'd have to define how to load the data. Those arguments would
+go here. For more information on custom loading read <TODO>.
+
+Extract operations
+------------------
+
+The operations list
+^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    operations = [
+        ...
+    ]
+
+This is a list of special functions that implement all of the
+convert/unify/reshape/standardize stuff described earlier. The most useful
+functions are already written for you. You just have to invoke them
+appropriately.
+
+For more information about extract operation functions, read <TODO>.
+
+A value map operation with functional replacements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    value_map(
+        in_col="participant",
+        m={
+            r"PID(\d+)": lambda x: int(x),  # strip PID and 0-padding
+        },
+        out_col=CONCEPT.PARTICIPANT.ID
+    )
+
+This says "Use the ``participant`` column as input (in_col=), replace
+everything that matches the regular expression pattern ``^PID(\d+)$`` with just
+the captured part and remove the zero padding by running the captured part
+through the function ``lambda x: int(x)`` (m={...}), and then output the result
+to a ``CONCEPT.PARTICIPANT.ID`` column (out_col=)."
+
+The resulting intermediate output will look like:
+
+.. csv-table::
+    :header: "index", "<CONCEPT.PARTICIPANT.ID>"
+
+    "0", "1"
+    "1", "2"
+    "2", "3"
+    "3", "4"
+    "4", "5"
+    "5", "6"
+    "6", "7"
+    "7", "8"
+    "8", "9"
+
+``lambda x: int(x)`` could be replaced by just ``int``, since the two
+expressions are functionally equivalent (both single-argument functions that
+effectively strip the leading zeros).
+
+We could also have kept these IDs as they were and instead converted the
+mother/father IDs, but, in the absence of an overriding directive such as input
+from the investigators about their preferences, it doesn't really make a
+difference which way we choose.
+
+A value map operation with variable replacements
```

Comment:
I think we need a brief section before this next value map that describes the concept schema

```
Ok before moving on to the next operation, let's pause for a second and understand what CONCEPT.PARTICIPANT.ID is and where it came from ..

Understanding the Standard Concept Schema
-----------------------------------------------
At the most basic level, the Standard Concept Schema is just a set of standardized attributes used by the ingest library. Contextually, these attributes represent a standard set of things or concepts in the world and a standard set of attributes for those things or concepts. The ideas is to transform the source data to this standardized intermediate set of values and then transform from the intermediate values to a set of target values defined by the target database or service.

So for example the for a column named `Body site` in our source data, we need to tell the ingest library what this means by specifying which concept and attribute this column name translates to. In this case, the column would probably map to `CONCEPT.BIOSPECIMEN.ANATOMICAL_SITE`. 

In the case of our value map above, we are mapping the `participant` column to `CONCEPT.PARTICIPANT.ID`, something that means the unique identifier for the standard concept of participant.  
```

Reply:
Actually, should we have a separate section outside of extract for the standard concept schema?

Reply:
somewhere else